File: day1/day1.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <algorithm>
#include <ranges>
using namespace std;

int main() {
    // std::cout << "hello world" << std::endl;
    std::ifstream input("large.input");
    std::string input_line;

    int left, right;
    std::vector<int> list1;
    std::vector<int> list2;

 
    while (input >> left >> right){
        list1.push_back(left);
        list2.push_back(right);
    }

    std::sort(list1.begin(), list1.end());
    std::sort(list2.begin(), list2.end());

    int sum = 0;

    for (size_t i = 0; i < list1.size(); ++i){
        sum += abs(list1[i] - list2[i]);
    }

    std::cout << "abs diff: " << sum << std::endl;

    // Part 2
    int similarity = 0;
    for (int i : list1){
        similarity += i * std::count(list2.begin(), list2.end(), i);
    }

    std::cout << "similarity: " << similarity << std::endl;
    
    return 0;
}

File: day1/day1.py



with open("large.input", "r") as f:
    l1 = []
    l2 = []
    for line in f:
        i1, i2 = line.split()
        l1.append(int(i1))
        l2.append(int(i2))
    
    l1.sort()
    l2.sort()

    total = 0
    for num1, num2 in zip(l1, l2):
        total += abs(num1 - num2)
    
    print(total)

    similarity = 0
    for num in l1:
        similarity += num * l2.count(num)
    
    print(similarity)

